src/main/haskell/language-kore/src/Data/Kore/Parser/ParserImpl.hs

```diff
@@ -599,12 +585,14 @@ BNF definitions:
     | ⟨meta-head⟩ ‘(’ ⟨pattern-list⟩ ‘)’
 @
 -}
-unifiedVariableOrTermPatternParser :: Parser UnifiedPattern
+unifiedVariableOrTermPatternParser :: Parser KorePattern
 unifiedVariableOrTermPatternParser = do
     c <- ParserUtils.peekChar'
     if c == '#'
-        then MetaPattern <$> variableOrTermPatternParser Meta
-        else ObjectPattern <$> variableOrTermPatternParser Object
+        then Fix . UnifiedPattern . UnifiedMeta . PatternObjectMeta
```

Comment:
This seems a bit ugly, why not 'asKorePattern'?

The same in the other places where this happens.